File: app/src/main/java/com/qadam/lastpuff/ui/viewmodel/AppViewModel.kt
package com.qadam.lastpuff.ui.viewmodel

import androidx.lifecycle.ViewModel
import androidx.lifecycle.ViewModelProvider
import androidx.lifecycle.viewModelScope
import com.qadam.lastpuff.data.repository.UserRepository
import com.qadam.lastpuff.domain.model.Achievement
import com.qadam.lastpuff.domain.model.HealthMilestone
import com.qadam.lastpuff.domain.model.HomeStats
import com.qadam.lastpuff.domain.model.MoneyGoal
import com.qadam.lastpuff.domain.model.ProgressStats
import com.qadam.lastpuff.domain.model.RecoveryIndex
import com.qadam.lastpuff.domain.model.SosContact
import com.qadam.lastpuff.domain.model.UserProfile
import com.qadam.lastpuff.domain.model.VictoryRecord
import com.qadam.lastpuff.domain.support.MessageSession
import com.qadam.lastpuff.domain.support.SosComposeFlags
import com.qadam.lastpuff.domain.support.SosMessageComposer
import com.qadam.lastpuff.domain.support.SosMode
import com.qadam.lastpuff.domain.support.SupportContext
import com.qadam.lastpuff.util.AppConstants
import com.qadam.lastpuff.util.StatsCalculator
import kotlinx.coroutines.delay
import kotlinx.coroutines.flow.MutableStateFlow
import kotlinx.coroutines.flow.SharingStarted
import kotlinx.coroutines.flow.StateFlow
import kotlinx.coroutines.flow.asStateFlow
import kotlinx.coroutines.flow.flow
import kotlinx.coroutines.flow.stateIn
import kotlinx.coroutines.launch
import java.util.Calendar

class AppViewModel(private val repository: UserRepository) : ViewModel() {

    private val tickFlow = flow {
        while (true) {
            emit(System.currentTimeMillis())
            delay(60_000)
        }
    }

    private var messageSession: MessageSession? = null

    val onboardingCompleted = repository.onboardingCompleted
        .stateIn(viewModelScope, SharingStarted.WhileSubscribed(5000), false)

    val darkTheme = repository.darkTheme
        .stateIn(viewModelScope, SharingStarted.WhileSubscribed(5000), null)

    val profile = repository.observeProfile()
        .stateIn(viewModelScope, SharingStarted.WhileSubscribed(5000), null)

    val sosContact = repository.observeSosContact()
        .stateIn(viewModelScope, SharingStarted.WhileSubscribed(5000), null)

    val moneyGoal = repository.observeMoneyGoal()
        .stateIn(viewModelScope, SharingStarted.WhileSubscribed(5000), null)

    val homeStats: StateFlow<HomeStats?> = repository.observeHomeStats(tickFlow)
        .stateIn(viewModelScope, SharingStarted.WhileSubscribed(5000), null)

    val progressStats: StateFlow<ProgressStats?> = repository.observeProgressStats(tickFlow)
        .stateIn(viewModelScope, SharingStarted.WhileSubscribed(5000), null)

    val healthMilestones: StateFlow<List<HealthMilestone>> = repository.observeHealthMilestones(tickFlow)
        .stateIn(viewModelScope, SharingStarted.WhileSubscribed(5000), emptyList())

    val achievements: StateFlow<List<Achievement>> = repository.observeAchievements()
        .stateIn(viewModelScope, SharingStarted.WhileSubscribed(5000), emptyList())

    val victories: StateFlow<List<VictoryRecord>> = repository.observeVictories()
        .stateIn(viewModelScope, SharingStarted.WhileSubscribed(5000), emptyList())

    val recoveryIndex: StateFlow<RecoveryIndex> = repository.observeRecoveryIndex(tickFlow)
        .stateIn(viewModelScope, SharingStarted.WhileSubscribed(5000), RecoveryIndex())

    val personalLetter = repository.personalLetter
        .stateIn(viewModelScope, SharingStarted.WhileSubscribed(5000), null)

    val notificationsEnabled = repository.notificationsEnabled
        .stateIn(viewModelScope, SharingStarted.WhileSubscribed(5000), true)

    val notificationHour = repository.notificationHour
        .stateIn(viewModelScope, SharingStarted.WhileSubscribed(5000), 9)

    val notificationMinute = repository.notificationMinute
        .stateIn(viewModelScope, SharingStarted.WhileSubscribed(5000), 0)

    val familyPhotoUri = repository.familyPhotoUri
        .stateIn(viewModelScope, SharingStarted.WhileSubscribed(5000), null)

    private val _sosState = MutableStateFlow(SosUiState())
    val sosState = _sosState.asStateFlow()

    fun completeOnboarding(profile: UserProfile, sosContact: SosContact, personalLetter: String?) {
        viewModelScope.launch {
            repository.completeOnboarding(profile, sosContact, personalLetter)
        }
    }

    fun updateProfile(profile: UserProfile) {
        viewModelScope.launch { repository.saveProfile(profile) }
    }

    fun updateSosContact(contact: SosContact) {
        viewModelScope.launch { repository.saveSosContact(contact) }
    }

    fun updatePersonalLetter(letter: String?) {
        viewModelScope.launch { repository.setPersonalLetter(letter) }
    }

    fun saveMoneyGoal(goal: MoneyGoal) {
        viewModelScope.launch { repository.saveMoneyGoal(goal) }
    }

    fun updateLastSmokeDate(timestamp: Long) {
        viewModelScope.launch { repository.updateLastSmokeDate(timestamp) }
    }

    fun setDarkTheme(enabled: Boolean) {
        viewModelScope.launch { repository.setDarkTheme(enabled) }
    }

    fun setNotificationsEnabled(enabled: Boolean) {
        viewModelScope.launch { repository.setNotificationsEnabled(enabled) }
    }

    fun setNotificationTime(hour: Int, minute: Int) {
        viewModelScope.launch { repository.setNotificationTime(hour, minute) }
    }

    fun setFamilyPhotoUri(uri: String?) {
        viewModelScope.launch { repository.setFamilyPhotoUri(uri) }
    }

    fun resetProgress() {
        viewModelScope.launch { repository.resetProgress() }
    }

    fun startSos() {
        messageSession = MessageSession()
        val mode = SosMode.forDay(Calendar.getInstance().get(Calendar.DAY_OF_YEAR))
        _sosState.value = SosUiState(
            step = SosStep.TIMER,
            timerSecondsLeft = AppConstants.SOS_TIMER_SECONDS,
            elapsedSeconds = 0,
            sosMode = mode
        )
        applySosMoment(0)
    }

    fun startEmergencySos() {
        messageSession = MessageSession()
        _sosState.value = SosUiState(step = SosStep.EMERGENCY, isEmergency = true)
    }

    fun startTimerFromEmergency() {
        val mode = SosMode.forDay(Calendar.getInstance().get(Calendar.DAY_OF_YEAR))
        if (messageSession == null) messageSession = MessageSession()
        _sosState.value = _sosState.value.copy(
            step = SosStep.TIMER,
            timerSecondsLeft = AppConstants.SOS_TIMER_SECONDS,
            elapsedSeconds = 0,
            sosMode = mode
        )
        applySosMoment(0)
    }

    fun resumeTimer() {
        val elapsed = _sosState.value.elapsedSeconds
        _sosState.value = _sosState.value.copy(step = SosStep.TIMER)
        applySosMoment(elapsed)
    }

    fun startBreathing() {
        _sosState.value = _sosState.value.copy(step = SosStep.BREATHING, breathingPhase = 0)
    }

    fun showOptionalDetails() {
        _sosState.value = _sosState.value.copy(showOptionalDetails = true)
    }

    fun dismissOptionalDetails() {
        _sosState.value = _sosState.value.copy(showOptionalDetails = false)
    }

    fun setOptionalTrigger(trigger: String) {
        _sosState.value = _sosState.value.copy(trigger = trigger)
    }

    fun setOptionalIntensity(intensity: Int) {
        _sosState.value = _sosState.value.copy(intensity = intensity)
    }

    fun tickTimer() {
        val state = _sosState.value
        if (state.step != SosStep.TIMER) return
        if (state.timerSecondsLeft <= 1) {
            _sosState.value = state.copy(step = SosStep.RESULT, timerSecondsLeft = 0)
        } else {
            val newSeconds = state.timerSecondsLeft - 1
            val elapsed = AppConstants.SOS_TIMER_SECONDS - newSeconds
            _sosState.value = state.copy(timerSecondsLeft = newSeconds, elapsedSeconds = elapsed)
            if (shouldUpdateSosMessage(elapsed)) {
                applySosMoment(elapsed)
            }
        }
    }

    private fun shouldUpdateSosMessage(elapsed: Int): Boolean =
        elapsed == 0 || elapsed % AppConstants.SOS_MESSAGE_INTERVAL_SECONDS == 0

    private fun applySosMoment(elapsed: Int) {
        val session = messageSession ?: return
        val state = _sosState.value
        val moment = SosMessageComposer.compose(
            elapsedSeconds = elapsed,
            ctx = buildSupportContext(),
            session = session,
            mode = state.sosMode,
            flags = SosComposeFlags(state.factShown, state.letterShown)
        )
        _sosState.value = state.copy(
            currentMessage = moment.primary,
            secondaryMessage = moment.secondary,
            currentAction = moment.action,
            showPersonalLetterDialog = moment.showPersonalLetter && !state.letterShown,
            factShown = state.factShown || (elapsed == 30 && moment.secondary != null),
            letterShown = state.letterShown || moment.showPersonalLetter
        )
    }

    private fun buildSupportContext(): SupportContext {
        val p = profile.value
        val goal = moneyGoal.value
        val pricePerCig = if (p != null && p.cigarettesInPack > 0) {
            p.packPrice / p.cigarettesInPack
        } else 0.0
        return SupportContext(
            reasons = p?.reasons ?: emptyList(),
            goalTitle = goal?.title,
            currency = p?.currency ?: "₸",
            pricePerCigarette = pricePerCig,
            totalWins = victories.value.size,
            personalLetter = personalLetter.value,
            hourOfDay = Calendar.getInstance().get(Calendar.HOUR_OF_DAY)
        )
    }

    fun tickBreathing() {
        val state = _sosState.value
        if (state.step != SosStep.BREATHING) return
        _sosState.value = state.copy(breathingPhase = state.breathingPhase + 1)
    }

    fun dismissPersonalLetter() {
        _sosState.value = _sosState.value.copy(
            showPersonalLetterDialog = false,
            letterShown = true
        )
    }

    fun showPhoto() {
        _sosState.value = _sosState.value.copy(showPhoto = true)
    }

    fun dismissPhoto() {
        _sosState.value = _sosState.value.copy(showPhoto = false)
    }

    fun showReasons() {
        _sosState.value = _sosState.value.copy(showReasons = true)
    }

    fun dismissReasons() {
        _sosState.value = _sosState.value.copy(showReasons = false)
    }

    fun completeSos(success: Boolean) {
        val state = _sosState.value
        val session = messageSession
        viewModelScope.launch {
            val recoveryPair = if (success) {
                repository.applySosRecoveryBoost()
            } else {
                RecoveryIndex() to RecoveryIndex()
            }
            repository.recordCraving(
                intensity = state.intensity,
                trigger = state.trigger.ifBlank { "Не указано" },
                success = success
            )
            val endMessage = if (success && session != null) {
                SosMessageComposer.victoryMessage(session)
            } else if (!success && session != null) {
                SosMessageComposer.relapseMessage(session)
            } else ""
            _sosState.value = state.copy(
                step = if (success) SosStep.SUCCESS else SosStep.RELAPSE,
                success = success,
                showCoinAnimation = success,
                currentMessage = endMessage,
                recoveryBefore = recoveryPair.first,
                recoveryAfter = recoveryPair.second
            )
        }
    }

    fun dismissCoinAnimation() {
        _sosState.value = _sosState.value.copy(showCoinAnimation = false)
    }

    fun resetSos() {
        messageSession = null
        _sosState.value = SosUiState()
    }

    fun moneySavedNow(): Double {
        val p = profile.value ?: return 0.0
        return StatsCalculator.moneySaved(p)
    }

    fun quitReasons(): List<String> = profile.value?.reasons ?: emptyList()

    fun liveHealthNow(): Pair<String, String> {
        val profile = profile.value ?: return "Сейчас" to "Начни путь — и тело сразу начнёт восстанавливаться."
        val elapsed = StatsCalculator.elapsedMillis(profile.lastSmokeDate)
        return AppConstants.liveHealthMessage(elapsed)
    }
}

enum class SosStep {
    EMERGENCY, TIMER, BREATHING, RESULT, SUCCESS, RELAPSE
}

data class SosUiState(
    val step: SosStep = SosStep.TIMER,
    val isEmergency: Boolean = false,
    val intensity: Int = 5,
    val trigger: String = "",
    val timerSecondsLeft: Int = AppConstants.SOS_TIMER_SECONDS,
    val elapsedSeconds: Int = 0,
    val currentMessage: String = "",
    val secondaryMessage: String? = null,
    val currentAction: String? = null,
    val sosMode: SosMode = SosMode.CHALLENGE,
    val breathingPhase: Int = 0,
    val factShown: Boolean = false,
    val letterShown: Boolean = false,
    val showOptionalDetails: Boolean = false,
    val showReasons: Boolean = false,
    val showPhoto: Boolean = false,
    val showPersonalLetterDialog: Boolean = false,
    val showCoinAnimation: Boolean = false,
    val success: Boolean = false,
    val recoveryBefore: RecoveryIndex = RecoveryIndex(),
    val recoveryAfter: RecoveryIndex = RecoveryIndex()
)

class AppViewModelFactory(private val repository: UserRepository) : ViewModelProvider.Factory {
    @Suppress("UNCHECKED_CAST")
    override fun <T : ViewModel> create(modelClass: Class<T>): T {
        if (modelClass.isAssignableFrom(AppViewModel::class.java)) {
            return AppViewModel(repository) as T
        }
        throw IllegalArgumentException("Unknown ViewModel class")
    }
}

File: ios/QadamLastPuff/ViewModels/AppViewModel.swift
import Foundation
import SwiftUI
import Combine

@MainActor
final class AppViewModel: ObservableObject {
    private let repository: UserRepository
    private var messageSession: MessageSession?
    private var minuteTimer: Timer?

    @Published var onboardingCompleted: Bool = false
    @Published var isFirstLaunch: Bool = true
    @Published var darkTheme: Bool?
    @Published var profile: UserProfile?
    @Published var sosContact: SosContact?
    @Published var moneyGoal: MoneyGoal?
    @Published var completedGoals: [CompletedMoneyGoal] = []
    @Published var homeStats: HomeStats?
    @Published var progressStats: ProgressStats?
    @Published var healthMilestones: [HealthMilestone] = []
    @Published var achievements: [Achievement] = []
    @Published var victories: [VictoryRecord] = []
    @Published var recoveryIndex = RecoveryIndex()
    @Published var personalLetter: String?
    @Published var notificationsEnabled: Bool = true
    @Published var notificationHour: Int = 9
    @Published var notificationMinute: Int = 0
    @Published var familyPhotoUri: String?
    @Published var sosState = SosUiState()

    init(repository: UserRepository) {
        self.repository = repository
        refreshAll()
        startMinuteTimer()
        repository.objectWillChange
            .receive(on: RunLoop.main)
            .sink { [weak self] _ in self?.refreshAll() }
            .store(in: &cancellables)
    }

    private var cancellables = Set<AnyCancellable>()

    private func startMinuteTimer() {
        minuteTimer?.invalidate()
        minuteTimer = Timer.scheduledTimer(withTimeInterval: 60, repeats: true) { [weak self] _ in
            Task { @MainActor in self?.refreshComputed() }
        }
    }

    func refreshAll() {
        onboardingCompleted = repository.onboardingCompleted
        isFirstLaunch = repository.isFirstLaunch
        darkTheme = repository.preferences.darkTheme
        profile = repository.profile
        sosContact = repository.sosContact
        moneyGoal = repository.moneyGoal
        completedGoals = repository.completedGoals
        personalLetter = repository.preferences.personalLetter
        notificationsEnabled = repository.preferences.notificationsEnabled
        notificationHour = repository.preferences.notificationHour
        notificationMinute = repository.preferences.notificationMinute
        familyPhotoUri = repository.preferences.familyPhotoUri
        refreshComputed()
    }

    func refreshComputed() {
        homeStats = repository.buildHomeStats()
        progressStats = repository.buildProgressStats()
        healthMilestones = repository.buildHealthMilestones()
        achievements = repository.buildAchievements()
        victories = repository.buildVictories()
        recoveryIndex = repository.computeRecoveryIndex()
    }

    func syncLaunchState() {
        repository.syncLaunchState()
        refreshAll()
    }

    func completeOnboarding(profile: UserProfile, sosContact: SosContact, personalLetter: String?) {
        repository.completeOnboarding(profile: profile, sosContact: sosContact, personalLetter: personalLetter)
        refreshAll()
    }

    func completeFirstLaunch() {
        repository.isFirstLaunch = false
        isFirstLaunch = false
    }

    func updateProfile(_ profile: UserProfile) {
        repository.saveProfile(profile)
        refreshAll()
    }

    func updateSosContact(_ contact: SosContact) {
        repository.saveSosContact(contact)
        refreshAll()
    }

    func updatePersonalLetter(_ letter: String?) {
        repository.preferences.personalLetter = letter
        personalLetter = letter
    }

    func saveMoneyGoal(_ goal: MoneyGoal) {
        repository.saveMoneyGoal(goal)
        refreshAll()
    }

    func archiveCompletedGoal() {
        repository.archiveCompletedGoal()
        refreshAll()
    }

    func isGoalReached() -> Bool {
        repository.checkGoalReached()
    }

    func moneySavedNow() -> Double {
        repository.moneySavedNow()
    }

    func updateLastSmokeDate(_ date: Date) {
        repository.updateLastSmokeDate(date)
        refreshAll()
    }

    func setDarkTheme(_ enabled: Bool) {
        repository.preferences.darkTheme = enabled
        darkTheme = enabled
    }

    func setNotificationsEnabled(_ enabled: Bool) {
        repository.preferences.notificationsEnabled = enabled
        notificationsEnabled = enabled
        NotificationManager.shared.reschedule(enabled: enabled, hour: notificationHour, minute: notificationMinute)
    }

    func setNotificationTime(hour: Int, minute: Int) {
        repository.preferences.notificationHour = hour
        repository.preferences.notificationMinute = minute
        notificationHour = hour
        notificationMinute = minute
        if notificationsEnabled {
            NotificationManager.shared.reschedule(enabled: true, hour: hour, minute: minute)
        }
    }

    func setFamilyPhotoUri(_ uri: String?) {
        repository.preferences.familyPhotoUri = uri
        familyPhotoUri = uri
    }

    func resetProgress() {
        repository.resetProgress()
        refreshAll()
    }

    func startSos() {
        messageSession = MessageSession()
        let dayOfYear = Calendar.current.ordinality(of: .day, in: .year, for: Date()) ?? 1
        sosState = SosUiState(
            step: .intake,
            sosMode: SosMode.forDay(dayOfYear: dayOfYear)
        )
    }

    func confirmSosIntakeAndStartTimer() {
        let dayOfYear = Calendar.current.ordinality(of: .day, in: .year, for: Date()) ?? 1
        sosState.step = .timer
        sosState.timerSecondsLeft = AppConstants.sosTimerSeconds
        sosState.elapsedSeconds = 0
        sosState.sosMode = SosMode.forDay(dayOfYear: dayOfYear)
        applySosMoment(0)
    }

    func setSosTrigger(_ trigger: String) { sosState.trigger = trigger }
    func setSosLocation(_ location: String) { sosState.location = location }
    func setSosIntensity(_ intensity: Int) { sosState.intensity = intensity }

    func showSosExitDialog() { sosState.showExitDialog = true }
    func dismissSosExitDialog() { sosState.showExitDialog = false }

    func exitSosWithoutTimer() {
        resetSos()
    }

    func restartSosTimer() {
        sosState.timerSecondsLeft = AppConstants.sosTimerSeconds
        sosState.elapsedSeconds = 0
        sosState.factShown = false
        sosState.letterShown = false
        applySosMoment(0)
        dismissSosExitDialog()
    }

    func startEmergencySos() {
        messageSession = MessageSession()
        sosState = SosUiState(step: .emergency, isEmergency: true)
    }

    func startTimerFromEmergency() {
        let dayOfYear = Calendar.current.ordinality(of: .day, in: .year, for: Date()) ?? 1
        if messageSession == nil { messageSession = MessageSession() }
        sosState.step = .intake
        sosState.sosMode = SosMode.forDay(dayOfYear: dayOfYear)
    }

    func resumeTimer() {
        sosState.step = .timer
        applySosMoment(sosState.elapsedSeconds)
    }

    func startBreathing() {
        sosState.step = .breathing
        sosState.breathingPhase = 0
    }

    func showOptionalDetails() { sosState.showOptionalDetails = true }
    func dismissOptionalDetails() { sosState.showOptionalDetails = false }
    func setOptionalTrigger(_ trigger: String) { sosState.trigger = trigger }
    func setOptionalIntensity(_ intensity: Int) { sosState.intensity = intensity }

    func tickTimer() {
        guard sosState.step == .timer else { return }
        if sosState.timerSecondsLeft <= 1 {
            sosState.step = .result
            sosState.timerSecondsLeft = 0
        } else {
            sosState.timerSecondsLeft -= 1
            sosState.elapsedSeconds = AppConstants.sosTimerSeconds - sosState.timerSecondsLeft
            if shouldUpdateSosMessage(sosState.elapsedSeconds) {
                applySosMoment(sosState.elapsedSeconds)
            }
        }
    }

    private func shouldUpdateSosMessage(_ elapsed: Int) -> Bool {
        elapsed == 0 || elapsed % AppConstants.sosMessageIntervalSeconds == 0
    }

    private func applySosMoment(_ elapsed: Int) {
        guard let session = messageSession else { return }
        let canShowLetter = repository.preferences.canShowPersonalLetterToday()
        let moment = SosMessageComposer.compose(
            elapsedSeconds: elapsed,
            ctx: buildSupportContext(),
            session: session,
            mode: sosState.sosMode,
            flags: SosComposeFlags(factShown: sosState.factShown, letterShown: sosState.letterShown)
        )
        sosState.currentMessage = moment.primary
        sosState.secondaryMessage = moment.secondary
        sosState.currentAction = moment.action
        let shouldShowLetter = moment.showPersonalLetter && !sosState.letterShown && canShowLetter
        sosState.showPersonalLetterDialog = shouldShowLetter
        if elapsed == 30 && moment.secondary != nil { sosState.factShown = true }
    }

    private func buildSupportContext() -> SupportContext {
        let p = profile
        let goal = moneyGoal
        let pricePerCig: Double
        if let p, p.cigarettesInPack > 0 {
            pricePerCig = p.packPrice / Double(p.cigarettesInPack)
        } else {
            pricePerCig = 0
        }
        return SupportContext(
            reasons: p?.reasons ?? [],
            goalTitle: goal?.title,
            currency: p?.currency ?? "₸",
            pricePerCigarette: pricePerCig,
            moneySaved: p.map { StatsCalculator.moneySaved(profile: $0) } ?? 0,
            totalWins: victories.count,
            personalLetter: personalLetter,
            hourOfDay: Calendar.current.component(.hour, from: Date())
        )
    }

    func tickBreathing() {
        guard sosState.step == .breathing else { return }
        sosState.breathingPhase += 1
    }

    func dismissPersonalLetter() {
        sosState.showPersonalLetterDialog = false
        sosState.letterShown = true
        repository.preferences.markPersonalLetterShownToday()
    }

    func showPhoto() { sosState.showPhoto = true }
    func dismissPhoto() { sosState.showPhoto = false }
    func showReasons() { sosState.showReasons = true }
    func dismissReasons() { sosState.showReasons = false }

    func completeSos(success: Bool) {
        let session = messageSession
        let recoveryPair = success ? repository.applySosRecoveryBoost() : (RecoveryIndex(), RecoveryIndex())
        repository.recordCraving(
            intensity: sosState.intensity,
            trigger: sosState.trigger.isEmpty ? "Не указано" : sosState.trigger,
            location: sosState.location,
            success: success
        )
        refreshAll()

        let endMessage: String
        if success, let session {
            endMessage = SosMessageComposer.victoryMessage(session: session)
        } else if !success, let session {
            endMessage = SosMessageComposer.relapseMessage(session: session)
        } else {
            endMessage = ""
        }

        sosState.step = success ? .success : .relapse
        sosState.success = success
        sosState.showCoinAnimation = success
        sosState.currentMessage = endMessage
        sosState.recoveryBefore = recoveryPair.0
        sosState.recoveryAfter = recoveryPair.1
    }

    func dismissCoinAnimation() { sosState.showCoinAnimation = false }

    func resetSos() {
        messageSession = nil
        sosState = SosUiState()
    }

    func quitReasons() -> [String] { profile?.reasons ?? [] }

    func liveHealthNow() -> (String, String) {
        guard let profile else {
            return ("Сейчас", "Начни путь — и тело сразу начнёт восстанавливаться.")
        }
        let elapsed = StatsCalculator.elapsedMillis(lastSmokeDate: profile.lastSmokeDate)
        return AppConstants.liveHealthMessage(elapsedMillis: elapsed)
    }
}
